tools/azure-sdk-tools/packaging_tools/auto_codegen.py

```diff
@@ -37,6 +40,57 @@ def init_new_service(package_name, folder_name):
             with open(str(ci), 'w') as file_out:
                 file_out.writelines(content)
 
+def update_servicemetadata(sdk_folder, data, config, folder_name, package_name, spec_folder, input_readme):
+
+    readme_file = str(Path(spec_folder, input_readme))
+    global_conf = config["meta"]
+    local_conf = config["projects"][readme_file]
+
+    cmd = ["autorest", input_readme]
+    cmd += build_autorest_options(global_conf, local_conf)
+
+    # metadata
+    metadata = {
+        "autorest": global_conf["autorest_options"]["version"],
+        "use": global_conf["autorest_options"]["use"],
+        "commit": data["headSha"],
+        "repository_url": data["repoHttpsUrl"],
+        "autorest_command": " ".join(cmd),
+        "readme": input_readme
+    }
+
+    _LOGGER.info("Metadata json:\n {}".format(json.dumps(metadata, indent=2)))
+
+    package_folder = Path(sdk_folder, folder_name, package_name).expanduser()
+    if not os.path.exists(package_folder):
+        _LOGGER.info(f"Package folder doesn't exist: {package_folder}")
+        _LOGGER.info("Failed to save metadata.")
+        return
+
+    metadata_file_path = os.path.join(package_folder, "_meta.json")
+    with open(metadata_file_path, "w") as writer:
+        json.dump(metadata, writer, indent=2)
+    _LOGGER.info(f"Saved metadata to {metadata_file_path}")
+
+    # Check whether MANIFEST.in includes _meta.json
+    require_meta = "include _meta.json\n"
+    manifest_file = os.path.join(package_folder, "MANIFEST.in")
+    if not os.path.exists(manifest_file):
```

Comment:
If MANIFEST.in doesn't exit, do we nee create one?

Reply:
I think we don't need to do it at this script. This file should be created from template. If not, it maybe something wrong when creating this package at first time.